moveit_core/collision_detection/src/world_diff.cpp

```diff
@@ -96,7 +96,7 @@ void WorldDiff::reset(const WorldPtr& world)
 void WorldDiff::setWorld(const WorldPtr& world)
 {
   WorldPtr old_world = world_.lock();
-  if (old_world)
+  if (old_world) // Variable world_ is encapsulated in the class, this is important. But not being a parameter it causes impure function
```

Comment:
Suggestion... for some reason, github is not letting me make a suggestion the normal way.

```
// Warning: function output depends on the class member `world_`. This makes it an "impure function" 
if (old_world)
```

Reply:
> Suggestion... for some reason, github is not letting me make a suggestion the normal way.
> 
> ```
> // Warning: function output depends on the class member `world_`. This makes it an "impure function" 
> if (old_world)
> ```

Thank you for your time! Shall I write all the comments in the future in this way?